### PR TITLE
PVAYLADEV-824

### PR DESCRIPTION
**Pull request by iluwatar**
* XML-JSON data conversion
* Elasticsearch data model
* Store environmental monitoring data to Elasticsearch
* Index and alias handling
* Elasticsearch cluster name, host and address are configurable

### CODE REVIEW

File: src/integration-test/java/fi/vrk/xroad/monitor/monitordata/MonitorDataHandlerTest.java [hunk 1]
```diff
@@ -61,8 +60,8 @@ public void makeRequest() {
         String xmlRequest = request.getRequestXML(exampleInfo);
         String root = handler.makeRequest(xmlRequest);
         log.info("result: {}", root);
-        String metric = response.getMetricInformation(root);
+        String metric = response.getMetricInformation(root, exampleInfo, "FI");
         log.info("body: {}", metric);
-        assertTrue(metric.contains("getSecurityServerMetricsResponse"));
+        Object jsonObject = JSONParser.parseJSON(metric);
```
Comment (Jevli) [inline src/integration-test/java/fi/vrk/xroad/monitor/monitordata/MonitorDataHandlerTest.java]: Should here be assert or somekind test?

File: src/main/java/fi/vrk/xroad/monitor/actor/MonitorDataActor.java [hunk 2]
```diff
@@ -60,32 +66,31 @@ public Receive createReceive() {
...
     }
 
-    protected void handleMonitorDataRequest(MonitorDataRequest request) {
-        log.info("start handleMonitorDataRequest {}", request.getSecurityServerInfo().toString());
-        String xml = handler.handleMonitorDataRequestAndResponse(request.getSecurityServerInfo());
-        log.info("Response metric: {}", xml);
-        if (xml.startsWith("<m:getSecurityServerMetricsResponse>")) {
-            saveMonitorData(xml, request.getSecurityServerInfo());
+    protected void handleMonitorDataRequest(MonitorDataRequest request)
+        throws ExecutionException, InterruptedException {
```
Comment (Jevli) [inline src/main/java/fi/vrk/xroad/monitor/actor/MonitorDataActor.java]: Exceptions are not handler, I think. Maybe theses should have been handled in Supervisor strategy.

File: src/main/java/fi/vrk/xroad/monitor/actor/MonitorDataActor.java [hunk 3]
```diff
@@ -60,32 +66,31 @@ public Receive createReceive() {
...
-     * Will save monitordata as json to elasticsearch. With securityserverinfo
-     * @param xml
+     * Saves monitordata as json to Elasticsearch
+     * @param json
      * @param securityServerInfo
      */
-    private void saveMonitorData(String xml, SecurityServerInfo securityServerInfo) {
-        // TODO create json element wchich has all data elasticsearch wants
-        // TODO make post to elasticsearch API
+    private void saveMonitorData(String json, SecurityServerInfo securityServerInfo)
```
Comment (Jevli) [inline src/main/java/fi/vrk/xroad/monitor/actor/MonitorDataActor.java]: Securityserverinfo is unused?

File: src/integration-test/java/fi/vrk/xroad/monitor/monitordata/MonitorDataHandlerTest.java (hunk 1 above)
Comment (iluwatar) [inline src/integration-test/java/fi/vrk/xroad/monitor/monitordata/MonitorDataHandlerTest.java]: Fixed

File: src/main/java/fi/vrk/xroad/monitor/actor/MonitorDataActor.java (hunk 2 above)
Comment (iluwatar) [inline src/main/java/fi/vrk/xroad/monitor/actor/MonitorDataActor.java]: Fixed

File: src/main/java/fi/vrk/xroad/monitor/actor/MonitorDataActor.java (hunk 3 above)
Comment (iluwatar) [inline src/main/java/fi/vrk/xroad/monitor/actor/MonitorDataActor.java]: Fixed

File: src/main/java/fi/vrk/xroad/monitor/actor/ResultCollectorActor.java [hunk 4]
```diff
@@ -58,8 +58,7 @@ public Receive createReceive() {
   }
```
Comment (jansu76) [inline src/main/java/fi/vrk/xroad/monitor/actor/ResultCollectorActor.java]: Actually not about this specific line, but the whole class....isnt the name "ResultCollectorActor" kind of ambiguous? MonitorDataActor also "collects" some kind of "results"? Maybe mention SecurityServerInfo or something more specific in the name to make the purpose more clear?

Comment (jansu76) [reply src/main/java/fi/vrk/xroad/monitor/actor/ResultCollectorActor.java]: Also name of ResultCollectorActor.MonitorDataResult is confusing to me, makes you think it is something MonitorDataActor produces?

Comment (jansu76) [inline src/main/java/fi/vrk/xroad/monitor/actor/ResultCollectorActor.java]: Actually not about this specific line, but the whole class....isnt the name "ResultCollectorActor" kind of ambiguous? MonitorDataActor also "collects" some kind of "results"? Maybe mention SecurityServerInfo or something more specific in the name to make the purpose more clear?

Also name of ResultCollectorActor.MonitorDataResult is confusing to me, makes you think it is something MonitorDataActor produces?

**Pull request merged**